pkg/notify/slack.go

```diff
@@ -59,32 +62,41 @@ func (s *Slack) SendEvent(event events.Event) error {
 	log.Debug(fmt.Sprintf(">> Sending to slack: %+v", event))
 	attachment := formatSlackMessage(event, s.NotifType)
 
-	// non empty value in event.channel demands redirection of events to a different channel
-	if event.Channel != "" {
-		channelID, timestamp, err := s.Client.PostMessage(event.Channel, slack.MsgOptionAttachments(attachment), slack.MsgOptionAsUser(true))
-		if err != nil {
-			log.Errorf("Error in sending slack message %s", err.Error())
-			// send error message to default channel
-			if err.Error() == "channel_not_found" {
-				msg := fmt.Sprintf("Unable to send message to Channel `%s`: `%s`\n```add Botkube app to the Channel %s\nMissed events follows below:```", event.Channel, err.Error(), event.Channel)
-				go s.SendMessage(msg)
-				// sending missed event to default channel
-				// reset event.Channel and send event
-				event.Channel = ""
-				go s.SendEvent(event)
-			}
-			return err
-		}
-		log.Debugf("Event successfully sent to channel %s at %s", channelID, timestamp)
-	} else {
+	targetChannel := event.Channel
+	if targetChannel == "" {
 		// empty value in event.channel sends notifications to default channel.
-		channelID, timestamp, err := s.Client.PostMessage(s.Channel, slack.MsgOptionAttachments(attachment), slack.MsgOptionAsUser(true))
-		if err != nil {
-			log.Errorf("Error in sending slack message %s", err.Error())
-			return err
+		targetChannel = s.Channel
+	}
+	isDefaultChannel := targetChannel == s.Channel
+
+	channelID, timestamp, err := s.Client.PostMessage(targetChannel, slack.MsgOptionAttachments(attachment), slack.MsgOptionAsUser(true))
+	if err != nil {
+		postMessageWrappedErr := fmt.Errorf("while posting message to channel %q: %w", targetChannel, err)
+
+		if isDefaultChannel || err.Error() != "channel_not_found" {
+			return postMessageWrappedErr
+		}
+
+		// channel not found, fallback to default channel
+
+		// send error message to default channel
+		msg := fmt.Sprintf("Unable to send message to Channel `%s`: `%s`\n```add Botkube app to the Channel %s\nMissed events follows below:```", targetChannel, err.Error(), targetChannel)
```

Comment:
The msg format can be declared as const. He can address that in separate PR as well

Reply:
This is not something I introduced, but will do that now 👍